src/Microsoft.OData.Core/ODataLibraryCompatibility.cs

```diff
@@ -4,29 +4,39 @@
 // </copyright>
 //---------------------------------------------------------------------
 
-using System.Diagnostics.CodeAnalysis;
+using System;
 
 namespace Microsoft.OData
 {
     /// <summary>
     /// Library compatibility levels.
     /// </summary>
-    [SuppressMessage("Microsoft.Design", "CA1008:EnumsShouldHaveZeroValue")]
-    public enum ODataLibraryCompatibility
+    [Flags]
+    public enum ODataLibraryCompatibility : long
     {
+        /// <summary>
+        /// No specific version or feature set.
+        /// </summary>
+        None = 0, // No flags set
+
+        /// <summary>
+        /// When enabled, the CSDL Scale and SRID "variable" value will be written "Variable" instead of "variable" for compatibility with older versions of the library.
+        /// </summary>
+        UseLegacyVariableCasing = 1 << 0,
+
         /// <summary>
         /// Version 6.x
         /// </summary>
-        Version6 = 060000,
+        Version6 = 1 << 4,
 
         /// <summary>
         /// Version 7.x
         /// </summary>
-        Version7 = 070000,
+        Version7 = 1 << 5,

```

Comment:
See my question here https://github.com/OData/odata.net/pull/2905/files#r1577262763

Reply:
@ElizabethOkerio How do you determine that `1 << 4` and `1 << 5` are large enough shifts to serve the purpose you mentioned? If that's your goal then perhaps you should assign them to the second largest and largest values respectively... Perhaps:
```csharp
Version6 = 1 << 61;
Version7 = 1 << 62;
```
Or allow some headroom above 7... (e.g. to accommodate Version8, etc in the future)
```csharp
Version6 = 1 << 56;
Version7 = 1 << 57;
```
Otherwise we'll only need to add 3 more compatibility flags to catch up with `1 << 4`.

Reply:
I don't think it's about the size it's about having them defined differently from the others..

Reply:
Assigning `1 << 4` and `1 << 5` is not defining them differently... Once we add
```csharp
P = 1 << 1,
Q = 1<< 2,
R = 1 << 3,
// .. Version6 & Version7
X = 1 << 6,
Y = 1 << 7,
Z = 1 << 8
```
will they still appear to be defined differently?